### ISSUE TITLE
trainGPU reuses the same observation buffer for both state and next_state

### ISSUE THREAD

**Issue by cerredz** [cerredz/Pulselib#22]
## Intent
`trainGPU.py` feeds a mutable observation buffer from `PokerGPU` directly into Q-learning. Because `PokerGPU.get_obs()` reuses `self.obs` and `step()` returns that same tensor again, the trainer can pass the same post-step buffer as both `states` and `next_states`, losing the pre-action observation entirely.

## Problem Description
Current behavior:
- `reset()` and `step()` both return `self.get_obs()`, which writes into and returns the shared `self.obs` tensor.
- `train_agent()` stores that tensor in `state`, then calls `env.step(actions)`.
- The subsequent `next_state` returned by `step()` aliases the same storage, so the original `state` reference is mutated in place before `train_step()` consumes it.

Expected behavior:
- The Q-learning update should receive an immutable snapshot of the pre-action state and a distinct snapshot of the post-action state.

Conditions:
- This manifests on every training update that relies on `state` surviving across a call to `env.step()`.
- It is a mainline training bug, not an edge case.

## Affected Files
- `environments/Poker/PokerGPU.py` - reuses a shared observation buffer and returns it from both `reset()` and `step()`.
- `scripts/Poker/trainGPU.py` - holds onto the returned tensor across `env.step()` and then passes it into `train_step()` without cloning.

## Code Snippets
```python
# environments/Poker/PokerGPU.py - lines 132-157
self.obs=torch.zeros((self.n_games, self.obs_size), device=self.device)
...
def get_obs(self):
    self.obs[:, 0:5] = self.board
    self.obs[:, 5:7] = self.hands[self.g, self.idx]
    ...
    self.obs[:, 15:end_idx:3] = self.current_round_bet[self.g.unsqueeze(1), opp_idx]

    return self.obs
```
`get_obs()` always returns the same `self.obs` tensor after mutating it in place.

```python
# environments/Poker/PokerGPU.py - lines 136 and 570
return self.get_obs(), self.get_info()
...
return self.get_obs(), rewards, self.is_done, self.is_truncated, self.get_info()
```
Both `reset()` and `step()` hand the shared observation buffer back to callers.

```python
# scripts/Poker/trainGPU.py - lines 33-37 and 50-60
state, info = env.reset(options={
    'rotation': rotations,
    'active_players': True,
    'q_agent_seat': q_seat
})
...
next_state, rewards, dones, truncated, info = env.step(actions)
...
agents[agent_types.index(PokerAgentType.QLEARNING)].train_step(
    states=state[active_games],
    actions=actions[active_games],
    rewards=rewards[active_games],
    next_states=next_state[active_games],
    dones=dones[active_games]
)
```
The trainer assumes `state` still holds the pre-step observation when `train_step()` runs.

## Root Cause Analysis
The root cause is buffer aliasing across the environment API boundary. `PokerGPU.get_obs()` writes into `self.obs` and returns it directly. `trainGPU.py` stores that tensor in `state`, but `env.step()` later mutates the same storage and returns it as `next_state`. By the time `train_step()` slices `state[active_games]`, `state` no longer contains the original observation. It has already been overwritten with the newest cursor state.

## Impact Assessment
- Affected flows: every Q-learning update in `trainGPU.py`.
- User impact: the network can be trained on `states` that no longer correspond to the action just taken.
- Data integrity risk: critical. This destroys the basic `(state, action, reward, next_state)` transition identity required for Q-learning.
- Failure mode: silent. Shapes and dtypes look valid, but the transition tuple is wrong.
- Severity: Critical. This corrupts the core supervised signal of the training loop.

## Potential Fixes
**Option A - Return cloned observations from `PokerGPU.reset()` and `step()`**
Change the environment API to return `self.get_obs().clone()` (or a freshly allocated tensor). This fixes aliasing at the source, but allocates more memory.

**Option B - Clone `state` and `next_state` in `trainGPU.py`**
Snapshot `state = state.clone()` before `env.step()` and `next_state = next_state.clone()` before training. This localizes the fix to the trainer, but other callers of `PokerGPU` could still be vulnerable.

**Option C - Add regression tests around storage identity**
Add a training-path test that asserts `state.data_ptr() != next_state.data_ptr()` across one step. This should ship with either code fix.

## Reproduction Steps
1. Instantiate `PokerGPU` with `n_players=3`, `max_players=3`, `n_games=1`.
2. Force seat `0` to act and assign distinct hole cards to seat `0` and seat `1`.
3. Call `state = env.get_obs()` and record `state[0, 5:7]`.
4. Call `next_state, ... = env.step(torch.tensor([1]))`.
5. Compare the buffers:
   - `state.data_ptr() == next_state.data_ptr()` is `True`
   - the original `state[0, 5:7]` value is overwritten with the post-step cards
6. In one direct repro:
   - pre-step snapshot: `[11, 24]`
   - `state` after `step()`: `[35, 48]`
   - `next_state`: `[35, 48]`

## Additional Context
- This issue compounds with any other next-state attribution bugs in the trainer because it invalidates the current-state side of the transition too.
- The environment tests currently focus on correctness of state contents, not identity/lifetime of the returned tensors.
